### PR TITLE
strings in objects API

### PR DESCRIPTION
**Pull request by Xavrax**
feat: possibility to use string in actions API

Added possibility to use strings in actions API

refactor: `pubnub_action_type` deprecated

`pubnub_action_type` enum has been deprecated

### CODE REVIEW

File: core/pbcc_actions_api.c [hunk 1]
```diff
@@ -17,31 +17,16 @@
 
 /* Maximum number of actions to return in response */
 #define MAX_ACTIONS_LIMIT 100
+#define MAX_ACTION_VALUE_LENGTH 15
```
Comment (parfeon) [inline core/pbcc_actions_api.c:20]:
```suggestion
#define MAX_ACTION_TYPE_LENGTH 15
```
Should it be like this?

File: core/pbcc_actions_api.c [hunk 2]
```diff
@@ -59,8 +44,35 @@ enum pubnub_res pbcc_form_the_action_object(struct pbcc_context* pb,
...
+        return PNR_INVALID_PARAMETERS;
+    }
+    if (('\"' != **val) || ('\"' != *(*val + pb_strnlen_s(*val, PUBNUB_MAX_OBJECT_LENGTH) - 1))) {
+        PUBNUB_LOG_ERROR("pbcc_form_the_action_object(pbcc=%p) - "
+                         "quotation marks on value ends are missing: "
+                         "value = %s\n",
+                         pb,
+                         *val);
+        return PNR_INVALID_PARAMETERS;
+    }
```
Comment (parfeon) [inline core/pbcc_actions_api.c:72]: If users now able to pass `string` as `type`, should we have the same check for it and ask users to enclose `type` into double quotes?

**Review by parfeon — APPROVED** [review]
Looks good but still some questions :)

File: core/pbcc_actions_api.c (hunk 1 above)
Comment (Xavrax) [reply core/pbcc_actions_api.c:20]: tbh, we should - commiting

File: core/pbcc_actions_api.c (hunk 2 above)
Comment (Xavrax) [reply core/pbcc_actions_api.c:72]: But wait. 
previous implementation haven't include the quotes. So in this case they shouldn't be forced to do such a thing ;o

File: core/pbcc_actions_api.c [hunk 3]
```diff
@@ -70,6 +70,14 @@ enum pubnub_res pbcc_form_the_action_object_str(struct pbcc_context* pb,
                          *val);
         return PNR_INVALID_PARAMETERS;
     }
+    if (('\"' != *action_type) || ('\"' != *(action_type + pb_strnlen_s(action_type, PUBNUB_MAX_OBJECT_LENGTH) - 1))) {
+        PUBNUB_LOG_ERROR("pbcc_form_the_action_object(pbcc=%p) - "
+                         "quotation marks on value ends are missing: "
```
Comment (parfeon) [inline core/pbcc_actions_api.c:75]: `"quotation marks on value ends are missing: "` -> `"quotation marks on action type ends are missing: "`

**Review by parfeon — APPROVED** [review]
LGTM!

**Comment by Xavrax** [comment]
@pubnub-release-bot release

**Comment by Xavrax** [comment]
@pubnub-release-bot release

**Comment by parfeon** [comment]
@pubnub-release-bot release

**Comment by parfeon** [comment]
@pubnub-release-bot release as v4.10.0

**Pull request merged**